contracts/core/UserOperationLib.sol

```diff
@@ -65,12 +64,24 @@ library UserOperationLib {
         return abi.encode(
             sender, nonce,
             hashInitCode, hashCallData,
-            callGasLimit, verificationGasLimit, preVerificationGas,
+            accountGasLimits, preVerificationGas,
             maxFeePerGas, maxPriorityFeePerGas,
             hashPaymasterAndData
         );
     }
 
+    function unpackAccountGasLimits(
+        bytes32 accountGasLimits
+    ) internal pure returns(uint128 validationGasLimit, uint128 callGasLimit) {
+        return (uint128(bytes16(accountGasLimits)), uint128(uint256(accountGasLimits)));
+    }
+
+    function unpackPaymasterStaticFields(
+        bytes calldata paymasterAndData
+    ) internal pure returns(address paymaster, uint128 validationGasLimit, uint128 postOp) {
+        return (address(bytes20(paymasterAndData[:20])), uint128(bytes16(paymasterAndData[20:36])), uint128(bytes16(paymasterAndData[36:52])));
```

Comment:
don't use offsets, extract to constants.
also, I think we should use this from entrypoint.

Reply:
Ok